main/src/ca/uwaterloo/flix/language/ast/DesugaredAst.scala

```diff
@@ -111,6 +111,8 @@ object DesugaredAst {
 
     case class LetRec(ident: Name.Ident, ann: Annotations, mod: Ast.Modifiers, exp1: Expr, exp2: Expr, loc: SourceLocation) extends Expr
 
+    case class LocalDef(ann: Annotations, ident: Name.Ident, fparams: List[FormalParam], declaredTpe: Option[Type], declaredEff: Option[Type], exp1: Expr, exp2: Expr, loc: SourceLocation) extends Expr
```

Comment:
I think, like with `Let`, that in these phases it should just be called `tpe` and `eff`